### PR TITLE
Update Hairsplitter

### PR DESCRIPTION
**Pull request by RolandFaure**
Update Hairsplitter to 1.9.9

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by RolandFaure** [comment]
@bioconda/core 
Hello,
This new updated version of the HairSplitter package does not pass the checks anymore.
The changes compared to the previous version that passed the tests were 1) a change of version (and of sha256sum, which I double checked) and 2) three new lines in the build.sh that do not seem suspicious to me
I tried building and installing the package locally on my computer and it seemed to work.
I am sorry if I missed something, I am not an expert in conda packages. Do you have any idea what could happen ?
Many thanks,
Roland

**Comment by bgruening** [comment]
@RolandFaure its failing only on OSX. What has changed in the upstream repo?

**Comment by martin-g** [comment]
I think the reason is the new version of `libcxx`. In the previous build (https://anaconda.org/bioconda/hairsplitter/1.9.5/download/osx-64/hairsplitter-1.9.5-hd8b7b98_0.tar.bz2) it used libcxx 17.x. Now it uses 18.x.
I pushed a commit. Let's see!

**Comment by martin-g** [comment]
The build failed with:
```
RuntimeError: SHA256 mismatch: '1eacdd1bf6a97a223c7a969ff9b77f1791e2866b1619c74f7badc89a9ce46dfe' != 'a539c9a1da1d4e96013aa6982c756192cc58038bf742f6d3a8f7c587ba7686d5'
```
@RolandFaure did you change the Git tag upstream ?!

File: recipes/hairsplitter/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% set name = "HairSplitter" %}
-{% set version = "1.9.5" %}
-{% set sha256 = "b05e9e6681056c18af7e136b3d537dac18a5ed24402612ca2173a4cced115c0b" %}
+{% set version = "1.9.9" %}
+{% set sha256 = "a539c9a1da1d4e96013aa6982c756192cc58038bf742f6d3a8f7c587ba7686d5" %}
```
Comment (martin-g) [inline recipes/hairsplitter/meta.yaml:3]:
```suggestion
{% set sha256 = "1eacdd1bf6a97a223c7a969ff9b77f1791e2866b1619c74f7badc89a9ce46dfe" %}
```

**Comment by RolandFaure** [comment]
Yes, sorry, I was trying to simplify the code and I did not see that the tests were running at the same time

**Comment by RolandFaure** [comment]
Thank you so much @bgruening and @martin-g , it would have taken me a lot of time to figure this out by myself <3

**Comment by RolandFaure** [comment]
@BiocondaBot please add label

**Review by martin-g — APPROVED** [review]

**Pull request merged**